Approved/implicitly-scoped-qubit-allocation.md

```diff
@@ -0,0 +1,314 @@
+   If no block is provided, the scope of the qubit is implicitly defined to be from the `use` or `borrow` statement until the end of the current block.
+2. Parentheses around the header are not required.
+   For consistency, parentheses are no longer required around any block statement header.
+
+Each example below is equivalent to the corresponding example in the [Current Status](#current-status) section with the same number.
+
+### Examples
+
+Example 1b: A single qubit allocated and released at the same time as the end of the outer block.
+
+```qsharp
+operation FlipCoin() : Result {
+    use q = Qubit();
+    H(q);
+    return M(q); // q is released here.
+}
+```
+
+Example 2b: Many qubits allocated in a single `use` statement with tuple destructuring.
+
+```qsharp
+operation QubitTuple(n : Int) : Result {
+    use (a, b, c) = (Qubit[2 * n + 1], Qubit[n], Qubit());
+    // ...
+    return M(c); // a, b, and c are released here.
+}
+```
+
+Example 3b: Many qubits allocated with a separate `use` statement for each variable.
+
+```qsharp
+operation NestedBlocks(n : Int) : Result {
+    use a = Qubit[2 * n + 1];
+    use b = Qubit[n];
+    use c = Qubit();
+    // ...
+    return M(c); // a, b, and c are released here.
+}
+```
+
+Example 5b: A clean qubit and a borrowed qubit. This can only be expressed with multiple statements, not with tuple destructuring.
+
+```qsharp
+operation UsingAndBorrowing() : Result {
+    use a = Qubit();
+    borrow b = Qubit();
+    // ...
+    return M(a); // a and b are released here.
+}
+```
+
+Example 5b: A qubit allocated within a new block and released before the end of the outer block.
+
+```qsharp
+operation DifferentLifetime() : Result {
+    mutable r = Zero;
+    use q = Qubit() {
+        // Apply operations...
+        set r = M(q);
+        // q is released here.
+    }
+    // Classical post-processing...
+    return r;
+}
+```
+
+# Implementation
+
+The `use` and `borrow` keywords can be added, and the `using` and `borrowing` keywords can be deprecated.
+Until `using` and `borrowing` are removed, they can be used in place of `use` and `borrow` for both the block and non-block statements.
+
+The block for the existing `using` and `borrowing` syntax nodes can be made optional.
+Either the block remains optional throughout all stages of the syntax tree, or the non-block form is transformed into the block form by automatically moving all statements below it into a new block.
+In the former case, all syntax tree consumers need to be aware of both forms, while in the latter case consumers would notice no change from the current syntax tree node.
+
+## Timeline
+
+There are no dependencies on other proposals or libraries.
+The work needed for either option appears to be minimal, since it is similar to the existing implementation.
+
+# Further Considerations
+
+Development tools should consider adding support for transitioning from the current syntax to the proposed syntax, such as a code action in IDEs.
```

Comment:
We should also gather telemetry to assess how often people choose to go with the block version. Let's discuss offline what data we need to gather to inform a possible future evolution for the block version.